statsmodels/tsa/base/tsa_model.py

```diff
@@ -120,7 +117,8 @@ def _get_predict_start(self, start):
         Start can be a string or an integer if self.data.dates is None.
         """
         dates = self.data.dates
-        if isinstance(start, str):
+        if not isinstance(start, (int, long)):
+            start = str(start)
```

Comment:
This is one fix I needed to make -- modern Pandas will output dates in various forms, e.g. a pandas `Period`